File: src/main/java/com/github/cannor147/request/Request.java
package com.github.cannor147.request;

import com.github.cannor147.configuration.Configuration;
import com.github.cannor147.model.Color;
import lombok.AccessLevel;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.util.Queue;

@Data
@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
public class Request {
    private final Queue<ColorizationTask> tasks;
    private final Configuration configuration;
    private final Color defaultColor;
}

File: src/main/java/com/github/cannor147/request/Request.kt
package com.github.cannor147.request

import com.github.cannor147.model.GeoMap
import com.github.cannor147.request.colorization.ColorizationTask
import java.util.*

class Request internal constructor(
    internal val tasks: Queue<ColorizationTask>,
    internal val geoMap: GeoMap
)